tools/fixFormat.js

```diff
@@ -1,104 +1,238 @@
+
+  function setNpmBasePaths(pkg) {
+    if (pkg.npmFileMap) {
+      for (var i in pkg.npmFileMap) {
+        var npmbasePath = pkg.npmFileMap[i].basePath;
+        if (npmbasePath && npmbasePath.length != 0 && ((typeof npmbasePath) == 'string')) {
+          npmbasePath = (npmbasePath).replace(/^\/+|\/+$/g , '');
           pkg.npmFileMap[i].basePath = npmbasePath;
+        }
       }
     }
   }
-  if (pkg.keywords !== undefined) {
-    var mod = _.uniq(pkg.keywords);
-    if (pkg.keywords.length != mod.length) {
-      pkg.keywords = mod;
+
+  function filterKeywords(pkg) {
+    if (pkg.keywords !== undefined) {
+      var mod = _.uniq(pkg.keywords);
+      if (pkg.keywords.length != mod.length) {
+        pkg.keywords = mod;
+      }
     }
   }
-  if (pkg.autoupdate) {
-    var basePath = pkg.autoupdate.basePath || ""
-    if (basePath || pkg.autoupdate.files) {
+
+  function fixAutoupdate(pkg) {
+    if (pkg.autoupdate) {
+      var basePath = pkg.autoupdate.basePath || ""
+      if (basePath || pkg.autoupdate.files) {
         if (basePath && basePath.length !== 0) {
-            basePath = basePath.replace(/^\/+|\/+$/g , "");
+          basePath = basePath.replace(/^\/+|\/+$/g , "");
         }
         pkg.autoupdate.fileMap = [
-            {
-                basePath: basePath || "",
-                files: pkg.autoupdate.files
-            }
+          {
+              basePath: basePath || "",
+              files: pkg.autoupdate.files
+          }
         ];
         delete pkg.autoupdate.basePath;
         delete pkg.autoupdate.files;
+      }
+    }
+  }
+
+  function updateJSON(path, pkg) {
+    fs.writeFile(path, JSON.stringify(pkg, null, 2) + '\n', 'utf8');
+    console.log(('Library - ' + pkg.name + ' been updated!').green);
+  }
+
+  function unRecognizedLicense(lib, license) {
+    console.log(('Library ' + lib + ' has un-recognized license - ' + license).yellow);
+  }
+
+  function adapt(licenseName) {
+    switch(licenseName) {
+      case 'GPLv2':
+      case 'GNU GPL v2':
+        return 'GPL-2.0';
+        break;
+      case 'GPLv3':
+      case 'GNU General Public License Version 3':
+        return 'GPL-3.0';
+        break;
+      case 'Apache License, Version 2.0':
+      case 'Apache 2.0':
+      case 'Apache License, 2.0':
+      case 'Apache License 2.0':
+        return 'Apache-2.0';
+        break;
     }
   }
-  fs.writeFileSync(item, JSON.stringify(pkg, null, 2) + '\n', 'utf8');
-  callback();
-});
+
+  function fixFilenameField(pkg, item) {
+    var orig = pkg.filename.split('.');
+    var min = '';
+    if (orig[orig.length - 2] !== 'min') {
+      var temp = orig,
+        ext = temp.pop();
+      temp.push('min');
+      temp.push(ext);
+      min = temp.join('.');
+    }
+    if (min !== '' && isThere('./ajax/libs/' + pkg.name + '/' + pkg.version + '/' + min)) {
+      pkg.filename = min;
+      fs.writeFileSync(item, JSON.stringify(pkg, null, 2) + '\n', 'utf8');
```

Comment:
Looks like we'll write the file at line 36, so we still need to write file here? Thanks.